src/siminterface/simulator.py

```diff
@@ -132,7 +143,8 @@ def apply(self, actions: SimulatorAction):
                                          self.sf_list, self.traffic, self.network_stats)
         self.writer.write_state_results(self.env, simulator_state)
         logger.debug(f"t={self.env.now}: {simulator_state}")
-
+        if self.params.use_states:
+            self.params.update_state()
```

Comment:
Wouldn't you need to call this on apply too?

Reply:
It is called at the end of apply, because we set a state in init, and we use that state for the first apply, then change it after wards for the next apply.

Reply:
Ah ok, then I just didn't see it there 👍